### PR TITLE
Fix PEP8 violations (lots!), pipe through wait_for_sync, and various login flow fixes/timing issues

### PR DESCRIPTION
**Pull request by jprouty**
Also cleans up a hairy section in json transactions where url params are hand crafted!

### CODE REVIEW

File: mintapi/api.py [hunk 1]
```diff
@@ -530,29 +550,32 @@ def get_transactions_json(self, include_investment=False,
...
         offset = 0
         # Mint only returns some of the transactions at once.  To get all of
         # them, we have to keep asking for more until we reach the end.
         while 1:
+            url = MINT_ROOT_URL + '/getJsonData.xevent'
+            params = {
+                'queryNew': '',
+                'offset': offset,
+                'comparableType': '8',
+                'rnd': Mint.get_rnd(),
```
Comment (mrooney) [inline mintapi/api.py]: This looks great! But what happens to query_options?

Comment (mrooney) [reply mintapi/api.py]: Ah I see, it ends up as params :)

Comment (jprouty) [reply mintapi/api.py]: Yup! the requests lib does all the url encoding instead :) Way cleaner and less error prone. This lib was already using params elsewhere, so this is nice to start unifying.

**Comment by jprouty** [comment]
Honestly, I'm not sure what's going on with the pep8 regressions! It was definitely working (a year ago?!) when I made my last PR. I'm happy to look a bit closer. Do you know if the modified travis.yaml modifies the current run? Or only the checked in travis is used?

**Comment by jprouty** [comment]
Oh; also worth mentioning: I verified everything is still working via accounts, budgets, and extended-transactions (since that was modified)

**Comment by jprouty** [comment]
Alright; this latest commit revives it. Guess who broke it 1 year ago... :) it's 'before_script', not 'before-script' :) My bad!

**Comment by jprouty** [comment]
In addition, **I've changed the py3 version to be 3.6**, which runs at the normal speed. For some reason, pandas (and therefore numpy) is freaking out on 3.4 and takes 10x the time the py2.7 and py3.6 test takes. I'd do 3.7, but for some reason that's not available on travis (probably older version of ubuntu or something).

This run illustrates all the versions together: https://travis-ci.org/mrooney/mintapi/builds/516678277

Should be all set for final review. Thanks!

**Comment by mrooney** [comment]
Beautiful, fantastic work, thanks so much for this great contribution :D

**Comment by jprouty** [comment]
My pleasure and apologies again for breaking the Travis CI :)

**Pull request merged**